nx/lib/nx.ex

```diff
@@ -4550,7 +4598,21 @@ defmodule Nx do
     %{tensor | shape: output_shape, names: output_names, vectorized_axes: []}
   end
 
-  def devectorize(tensor, _), do: tensor
+  def devectorize(tensor, _)
+      when tensor.vectorized_axes == []
+      when is_struct(tensor, Complex)
+      when is_number(tensor),
+      do: tensor
```

Comment:
Yes, just so it works better with containers. I don't want vectorize to be the same because we have the check for scalars there that we don't have here.

An option to consistency between both functions: we can open an exception on vectorize to plain numbers, though, but I would find that behavior a bit strange, or do to_tensor here.

Reply:
I'll merge this, but if you prefer one of the alternatives, I can send a separate PR on this